src/attack_flow_builder/src/assets/builder.config.validator.ts

```diff
@@ -203,11 +213,10 @@ class AttackFlowValidator extends DiagramValidator {
                     this.addError(id, "Invalid MIME Type.");
                 }
                 // Validate Payload Bin, URL, and Hashes
-                if(payloadBin?.isDefined() && url?.isDefined()) {
-                    this.addError(id, "Artifact must have either have a Payload Bin or URL, not both.");
-                }
-                if(url?.isDefined() && !hashes?.isDefined()) {
-                    this.addError(id, "Artifact URL must also have a Hash.");
+                if (payloadBin?.isDefined() && (url?.isDefined() || hashes?.isDefined()) ||
+                   (url?.isDefined() && !hashes?.isDefined()) ||
+                   (hashes?.isDefined() && !url?.isDefined())) {
```

Comment:
The [STIX Documentation](https://docs.oasis-open.org/cti/stix/v2.1/os/stix-v2.1-os.html#_4jegwl6ojbes) states that if `url` is present, `hashes` must be present. It never states that if `hashes` is present, `url` must be present.

Reply:
Ahh yeah I misread that. I'll fix it.